packages/task/src/node/task-server.ts

```diff
@@ -0,0 +1,232 @@
+                proc = this.terminalProcessFactory(<TerminalProcessOptions>processOptions);
+            } else {
+                this.logger.info('Task: creating underlying raw process');
+                proc = this.rawProcessFactory(<RawProcessOptions>processOptions);
+            }
+
+            task = new Task(
+                this.taskManager,
+                proc,
+                options.errorMatcherName
+            );
+
+            // when underlying process exits, notify tasks listeners
+            const toDispose = new DisposableCollection();
+
+            toDispose.push(
+                proc.onExit(event => {
+                    this.fireTaskExitedEvent({
+                        'taskId': task.id,
+                        'code': event.code,
+                        'signal': event.signal
+                    });
+                })
+            );
+
+            this.taskToDispose.set(task.id, toDispose);
+
+            return Promise.resolve(
+                {
+                    taskId: task.id,
+                    osProcessId: proc.pid,
+                    terminalId: (options.processType === 'terminal') ? proc.id : undefined,
+                    processId: proc.id
+                }
+            );
+
+        } catch (error) {
+            this.logger.error(`Error occured while creating task: ${error}`);
+            return Promise.reject(new Error(error));
+        }
+    }
+
+    protected fireTaskExitedEvent(event: ITaskExitedEvent) {
+        this.logger.debug(log => log(`task has exited:`, event));
+        if (this.client) {
+            this.client.onTaskExit(event);
+        }
+    }
+
+    public tasksListChanged(tasks: string[]) {
+        this.fireTaskChangedEvent({ 'tasks': tasks });
+    }
+
+    protected fireTaskChangedEvent(event: ITasksChangedEvent) {
+        this.logger.debug(log => log(`Configured tasks changed:`, event));
+        if (this.client) {
+            this.client.onTasksConfigChanged(event);
+        }
+    }
+
+    kill(id: number): Promise<void> {
+        const taskToKill = this.taskManager.get(id);
+        if (taskToKill !== undefined) {
+            taskToKill.kill();
+        }
+        return Promise.resolve();
+    }
+
+    /** Set the client we send the notifications-to */
+    setClient(client: ITaskClient | undefined) {
+        this.client = client;
+    }
+
+    // Check if task command can be found, if so return its resolved
+    // absolute path, else returns undefined
+    private findCommand(command: string, cwd: string): string | undefined {
+        const systemPath = process.env.PATH;
+
+        if (path.isAbsolute(command) && fs.existsSync(command)) {
+            return command;
+        } else if (path.isAbsolute(command)) {
+            const uri = new URI(cwd).resolve(command);
+            const resolved_command = FileUri.fsPath(uri);
+            if (fs.existsSync(resolved_command)) {
```

Comment:
don't use sync

Reply:
Use fs.access()

Reply:
I know in general it's better to go async, but do you see a practical reason here? No reason for it to block for long here

Reply:
With fs' sync-methods the whole process (backend server) will block until the file system returns. So we should never use it in the backend. 

See also https://stackoverflow.com/a/17605734/1035471

It is ok to use it in separated processes and/or scripts, though.

Reply:
I opened https://github.com/theia-ide/theia/issues/762

Reply:
Allright - thanks for the explanation. I will stop using the fs.*sync functions in this context

Reply:
Note a good solution is to use async/await this makes it easy to convert sync to async without modifing your code flow too much by addmin promises etc

Reply:
done - now using only async fs.access().